benchmark/JoinAlgorithmBenchmark.cpp

```diff
@@ -790,11 +790,10 @@ class GeneralInterfaceImplementation : public BenchmarkInterface {
       };
     };
     auto generateBiggerEqualLambdaDesc =
-        []<typename OptionType,
-           typename = std::enable_if_t<ad_utility::isInstantiation<
-               OptionType, ad_utility::ConstConfigOptionProxy>>>(
+        CPP_template_lambda()(typename OptionType)(
             const OptionType& option, const auto& minimumValue,
-            bool canBeEqual) {
+            bool canBeEqual)(requires ad_utility::isInstantiation<
+                             OptionType, ad_utility::ConstConfigOptionProxy>) {
```

Comment:
You cannot use the `CPP_template_lambda`  macro, as it uses the feature that you are trying to get rid of.
You can for example use `CPP_lambda` with explicit decltype statements, that should work for at least the non-variadic templates.

Reply:
Addressed